File: MiniDOMOffice/Office/GDE/GDE.vb
Imports minidom
Imports minidom.Databases
Imports minidom.Sistema
Imports minidom.Anagrafica
Imports minidom.Office
Imports minidom.Internals

Namespace Internals

    Public NotInheritable Class CGDECLass
        Inherits CModulesClass(Of CDocumento)

        Friend Sub New()
            MyBase.New("modGDE", GetType(CDocumentiCursor), -1)
        End Sub

        Public Function GetItemByName(ByVal name As String) As CDocumento
            name = Strings.Trim(name)
            If (name = "") Then Return Nothing
            For Each d As CDocumento In Me.LoadAll
                If Strings.Compare(d.Nome, name) = 0 Then Return d
            Next
            Return Nothing
        End Function


    End Class

End Namespace

Partial Class Office


    Private Shared m_GDE As Internals.CGDECLass = Nothing

    Public Shared ReadOnly Property GDE As CGDECLass
        Get
            If (m_GDE Is Nothing) Then m_GDE = New CGDECLass
            Return m_GDE
        End Get
    End Property

End Class

File: MiniDOMOffice/Office/GDE/GDE.cs
using System;
using System.Collections;
using System.Collections.Generic;
using DMD;
using DMD.XML;
using DMD.Databases;
using DMD.Databases.Collections;
using minidom;
using minidom.repositories;
using static minidom.Sistema;
using static minidom.Anagrafica;
using static minidom.Office;


namespace minidom
{
    namespace repositories
    {

        /// <summary>
        /// Repository di <see cref="DocumentoCaricato"/>
        /// </summary>
        [Serializable]
        public sealed class CGDECLass 
            : CModulesClass<DocumentoCaricato>
        {
            
            /// <summary>
            /// Costruttore
            /// </summary>
            public CGDECLass() 
                : base("modGDE", typeof(DocumentiCaricatiCursor), -1)
            {
            }

            /// <summary>
            /// Restituisce il documento in base al nome
            /// </summary>
            /// <param name="name"></param>
            /// <returns></returns>
            public minidom.Office.DocumentoCaricato GetItemByName(string name)
            {
                name = DMD.Strings.Trim(name);
                if (string.IsNullOrEmpty(name))
                    return null;
                foreach (minidom.Office.DocumentoCaricato d in LoadAll())
                {
                    if (DMD.Strings.Compare(d.Nome, name, true) == 0)
                        return d;
                }

                return null;
            }
        }
    }

    public partial class Office
    {
        private static CGDECLass m_GDE = null;

        /// <summary>
        /// Repository di <see cref="DocumentoCaricato"/>
        /// </summary>
        public static CGDECLass GDE
        {
            get
            {
                if (m_GDE is null)
                    m_GDE = new CGDECLass();
                return m_GDE;
            }
        }
    }
}